app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningActivity.kt

```diff
@@ -125,19 +146,25 @@ class LearningActivity : AppCompatActivity() {
         }
     }
 
+    /**
+     * This callback is called when the image to sort is dropped on a target ImageView
+     * @param event the DragEvent representing the interaction
+     * @param v The ImageView representing the target category
+     */
     private fun dropCallback(event: DragEvent, v: View): Boolean {
         val item: ClipData.Item = event.clipData.getItemAt(0)
-        v.alpha = opaque
-        v.invalidate()
-        Log.d("dropCallback", "${item.text} vs ${v.contentDescription}")
+        setOpacity(v, opaque)
+        Log.d(localClassName, "dropped : ${item.text} on category: ${v.contentDescription}")
+
+        // the classification is considered correct if the text carried by the drag
```

Comment:
That was always kind of a fragile hack right? Does it work with `FirestoreCategorizedPicture` now?

Reply:
I realized now that FirestoreCategorizedPicture breaks my usage of contentDescription.
Will be changed in next sprint